PAY-4845: Adds pages for the new queries and mutations introduced with payment services graphql functionality (#240)
AdobeDocs/commerce-webapi: src/pages/graphql/schema/checkout/queries/get-payment-config.md

---
title: getPaymentConfig query
---

# getPaymentConfig query

The `getPaymentConfig` query returns the payment configuration for the available payment methods for a specific `PaymentLocation`:

* CHECKOUT
* MINICART
* CART
* PRODUCT_DETAIL
* ADMIN

These payment methods for the `getPaymentConfig` query are the ones supported in [Payment Services](https://experienceleague.adobe.com/docs/commerce-merchant-services/payment-services/payments-checkout/payments-options.html):

* PayPal Hosted Fields
* Apple Pay
* PayPal Smart Buttons

## getPaymentConfig object

> 

Reviewer: Add a H2 Output attributes

> The `getPaymentConfig` object will differ on which attributes must contain depending on the payment method.

Reviewer: ```suggestion
The `PaymentConfigOutput` contains details about each configured payment method.
```

Reviewer: I can't correct the output information from the GH PR review boxes. We have to have a meeting where I show you how to assemble this info.

`is_vault_enabled` | Boolean | Indicates whether card vaulting is enabled. The possible values are `True` or `False`
`payment_source` | String! | Optional. The identifiable payment source for the payment method.
`requires_card_details` | Boolean | Indicates whether card and bin details are required. This value is true when the Signifyd integration is enabled for `hosted_fields`. The possible values are `True` or `False`
`three_ds` | Boolean | Indicates whether 3DS mode is enabled. The possible values are `True` or `False`

### `SmartButtonsConfig` attributes

The `SmartButtonsConfig` payment method configuration contains the following attributes:

Attribute |  Data Type | Description
--- | --- | ---
`button_styles` | String | The styles for the PayPal Smart Button configuration
`display_message` | Boolean | Defines if PayPal Pay Later message is shown. The possible values are `True` or `False`
`display_venmo` | Boolean | Defines if the Venmo option is shown. The possible values are `True` or `False`
`message_styles` | String | The message styles for the PayPal Pay Later configuration

> #### `button_styles` object

Reviewer: ```suggestion
#### `ButtonStyles` object
```

> The `SmartButtonsConfig` payment method configuration has a `button_styles` object containing the following attributes:

Reviewer: ```suggestion
The `SmartButtonsConfig` payment method configuration has a `ButtonStyles` object containing the following attributes:
```

Attribute | Data Type | Description
--- | --- | ---
`color` | String | The button color
`height` | Int | The button height
`label` | String | The button label
`layout` | String | The button layout
`shape` | String | The button shape
`tagline` | Boolean | Displays tagline. The possible values are `True` or `False`
`use_default_height` | Boolean | Defines if button uses default height. The possible values are `True` or `False`. If value is `False`, `height` is used

> #### `message_styles` object

Reviewer: ```suggestion
#### `MessageStyles ` object
```

> The `SmartButtonsConfig` payment method configuration has a `message_styles` object containing the following attributes:

Reviewer: ```suggestion
The `SmartButtonsConfig` payment method configuration has a `MessageStyles ` object containing the following attributes:
```

Attribute | Data Type | Description
--- | --- | ---
`layout` | String | The message layout
`logo` | String | The message logo

> #### `SDK_Params` attributes

Reviewer: ```suggestion
#### `SDKParams` attributes
```

> The `sdk_params` object provides details about the SDK parameters:

Reviewer: ```suggestion
The `SDKParams` object provides details about the SDK parameters:
```

State where these values are coming from.

## `PaymentConfigItem` interface

The `PaymentConfigItem` interface contains the fields that are common to all the payment methods. This interface contains the following attributes:

Attribute |  Data Type | Description
--- | --- | ---
`code` | String! | The payment method code as defined in the payment gateway
`is_visible` | Boolean | Indicates whether the payment method is shown
`payment_intent` | String | Defines the payment intent. The possible values are `Authorize` or `Capture`
`sdk_params` | SDKParams | PayPal parameters required to load the PayPal JavaScript SDK
`sort_order` | String | Defines the preference for the sorting order of the payment method
`title` | String! | The display name of the payment method

This interface also has the following implementations:

* ApplePayConfig
* HostedFieldsConfig
* SmartButtonsConfig

Each implementation contains objects with specific attributes for each of tthese payment methods.

### `ApplePayConfig` attributes

The `ApplePayConfig` payment method configuration has a `button_styles` object containing the following attributes:

Attribute | Data Type | Description
--- | --- | ---
`color` | String | The button color

> `height` | Units | The button height

Reviewer: ```suggestion
`height` | Int | The button height
```

By units, I meant "The button height, in pixels", or whatever the correct unit is.

Author: aaah, thanks

Author: Cannot find this information anywhere, I'll ask the team meanwhile

Author: great

#### `ButtonStyles` object

The `SmartButtonsConfig` payment method configuration has a `ButtonStyles` object containing the following attributes:

Attribute | Data Type | Description
--- | --- | ---
`color` | String | The button color
`height` | Int | The button height
`label` | String | The button label
`layout` | String | The button layout
`shape` | String | The button shape
`tagline` | Boolean | Indicates whether to display the PayPal tagline
`use_default_height` | Boolean | Defines if button uses default height. If the value is `False`, the value of `height` is used

#### `MessageStyles` object

The `SmartButtonsConfig` payment method configuration has a `MessageStyles` object containing the following attributes:

Attribute | Data Type | Description
--- | --- | ---
`layout` | String | The message layout
`logo` | String | The message logo

#### `SDKParams` attributes

The `SDKParams` object provides details about the SDK parameters:

Attribute |  Data Type | Description
--- | --- | ---
`name` | String! | The name of the SDK parameter

> `value` | String! | The value of the SDK parameter

Reviewer: ```suggestion
`value` | String | The value of the SDK parameter
```

`location` | PaymentLocation! | The origin location for that payment request. The possible values are PRODUCT_DETAIL, MINICART, CART, CHECKOUT, ADMIN

## Output attributes

The `PaymentConfigOutput` contains details about each configured payment method:

Attribute |  Data Type | Description
--- | --- | ---
`ApplePayConfig` | String! | ApplePay payment method configuration
`HostedFieldsConfig` | String! | PayPal Hosted fields payment method configuration
`SmartButtonsConfig` | String! | PayPal Smart Buttons payment method configuration

Each of these output attributes implements the `PaymentConfigItem` interface.

## `PaymentConfigItem` interface

The `PaymentConfigItem` interface contains the fields that are common to all the payment methods. This interface contains the following attributes:

Attribute |  Data Type | Description
--- | --- | ---
`code` | String | The payment method code as defined in the payment gateway
`is_visible` | Boolean | Indicates whether the payment method is shown
`payment_intent` | String | Defines the payment intent. The possible values are `Authorize` or `Capture`
`sdk_params` | SDKParams | PayPal parameters required to load the PayPal JavaScript SDK

> `sort_order` | String | Defines the preference for the sorting order of the payment method

Reviewer: ```suggestion
`sort_order` | String | The relative order the payment method is displayed on the checkout page
```
